src/shapes/text.class.js

```diff
@@ -1152,22 +1156,42 @@
     },
 
     /**
-     * Turns the character into an 'inferior figure' (aka. 'subscript')
-     * @param {Number} line the line number or starting position
-     * @param {Number} char the character number or final position
-     * @param {Object} schema either this.superscript or this.subscript
-     * @returns {Object} this
+     * Applies 'schema' to the character at given position
      * @private
+     * @param {Number} line the line number
+     * @param {Number} char the character number
+     * @param {Object} schema either this.superscript or this.subscript
+     * @returns {Object} thisArg
+     * @chainable
      */
     _setScript: function(line, char, schema) {
+      if (!this._checkLocation(line, char) || typeof schema !== 'object') {
+        return this;
+      }
+      if (typeof schema.size !== 'number' || typeof schema.baseline !== 'number') {
+        return this;
+      }
```

Comment:
We do not take care of this kind of checks.
If the dev provide a line/char that does not exist, he will set the style for that numbers.
There may be reasons to do that.